Cannot build on macOS

I'm following the instructions in the readme for macOS and I get the following error about a redundant semicolon in glm:

```
[  2%] Built target dbopl
[  6%] Built target entityx
[  8%] Built target glad
[ 10%] Built target speex_resampler
[ 10%] Building CXX object src/CMakeFiles/rigel_core.dir/engine/debugging_system.cpp.o
In file included from /Users/kylefalconer/sources/RigelEngine/src/engine/debugging_system.cpp:17:
In file included from /Users/kylefalconer/sources/RigelEngine/src/engine/debugging_system.hpp:22:
In file included from /Users/kylefalconer/sources/RigelEngine/src/engine/renderer.hpp:24:
In file included from /Users/kylefalconer/sources/RigelEngine/src/engine/shader.hpp:23:
In file included from /Users/kylefalconer/sources/RigelEngine/3rd_party/glm/glm/gtc/type_ptr.hpp:37:
In file included from /Users/kylefalconer/sources/RigelEngine/3rd_party/glm/glm/gtc/../gtc/quaternion.hpp:18:
In file included from /Users/kylefalconer/sources/RigelEngine/3rd_party/glm/glm/gtc/../gtc/matrix_transform.hpp:24:
In file included from /Users/kylefalconer/sources/RigelEngine/3rd_party/glm/glm/gtc/../mat4x4.hpp:5:
In file included from /Users/kylefalconer/sources/RigelEngine/3rd_party/glm/glm/gtc/.././ext/matrix_double4x4.hpp:5:
In file included from /Users/kylefalconer/sources/RigelEngine/3rd_party/glm/glm/gtc/../ext/../detail/type_mat4x4.hpp:188:
In file included from /Users/kylefalconer/sources/RigelEngine/3rd_party/glm/glm/gtc/../ext/../detail/type_mat4x4.inl:1:
In file included from /Users/kylefalconer/sources/RigelEngine/3rd_party/glm/glm/gtc/../ext/../detail/../matrix.hpp:161:
In file included from /Users/kylefalconer/sources/RigelEngine/3rd_party/glm/glm/gtc/../detail/func_matrix.inl:1:
In file included from /Users/kylefalconer/sources/RigelEngine/3rd_party/glm/glm/gtc/../ext/../detail/../geometric.hpp:116:
In file included from /Users/kylefalconer/sources/RigelEngine/3rd_party/glm/glm/gtc/../detail/func_geometric.inl:2:
In file included from /Users/kylefalconer/sources/RigelEngine/3rd_party/glm/glm/gtc/../ext/../detail/../common.hpp:532:
In file included from /Users/kylefalconer/sources/RigelEngine/3rd_party/glm/glm/gtc/../detail/func_common.inl:785:
In file included from /Users/kylefalconer/sources/RigelEngine/3rd_party/glm/glm/gtc/../ext/../detail/func_common_simd.inl:6:
/Users/kylefalconer/sources/RigelEngine/3rd_party/glm/glm/gtc/../ext/../detail/../simd/common.h:106:45: error: empty expression statement
      has no effect; remove unnecessary ';' to silence this warning [-Werror,-Wextra-semi-stmt]
        glm_vec4 const or0 = _mm_or_ps(and0, and1);;
                                                   ^
1 error generated.
make[2]: *** [src/CMakeFiles/rigel_core.dir/engine/debugging_system.cpp.o] Error 1
make[1]: *** [src/CMakeFiles/rigel_core.dir/all] Error 2
make: *** [all] Error 2
```

Once I modify the glm code to fix the issue, I get a new error:

```
...
[ 49%] Building CXX object src/CMakeFiles/rigel_core.dir/game_logic/game_world.cpp.o
In file included from /Users/kylefalconer/sources/RigelEngine/src/game_logic/game_world.cpp:28:
In file included from /Users/kylefalconer/sources/RigelEngine/src/loader/resource_loader.hpp:21:
/Users/kylefalconer/sources/RigelEngine/src/data/movie.hpp:27:3: error: explicitly defaulted default constructor is implicitly deleted
      [-Werror,-Wdefaulted-function-deleted]
  MovieFrame() = default;
  ^
/Users/kylefalconer/sources/RigelEngine/src/data/movie.hpp:34:9: note: default constructor of 'MovieFrame' is implicitly deleted because
      field 'mReplacementImage' has no default constructor
  Image mReplacementImage;
        ^
1 error generated.
make[2]: *** [src/CMakeFiles/rigel_core.dir/game_logic/game_world.cpp.o] Error 1
make[1]: *** [src/CMakeFiles/rigel_core.dir/all] Error 2
make: *** [all] Error 2
```

Adding `-Wextra-semi-stmt` to the `src/base/warnings.hpp` helped go past the warning, but another occurred, before reaching 49%, which mentions macOS 10.4 (I'm on 10.3.6).

I'm guessing Mojave is a minimum requirement then.

```
...
[ 32%] Building CXX object src/CMakeFiles/rigel_core.dir/game_logic/ai/bomber_plane.cpp.o
In file included from /Users/avioli/repos/RigelEngine/src/game_logic/ai/bomber_plane.cpp:19:
/Users/avioli/repos/RigelEngine/src/base/match.hpp:40:10: error: call to unavailable function 'visit': introduced in macOS 10.14
  return std::visit(
         ^~~~~~~~~~
/Users/avioli/repos/RigelEngine/src/game_logic/ai/bomber_plane.cpp:108:9: note: in instantiation of function template
      specialization 'rigel::base::match<std::__1::variant<rigel::game_logic::behaviors::BomberPlane::FlyingIn,
      rigel::game_logic::behaviors::BomberPlane::DroppingBomb, rigel::game_logic::behaviors::BomberPlane::FlyingOut> &, (lambda at
      /Users/avioli/repos/RigelEngine/src/game_logic/ai/bomber_plane.cpp:109:5), (lambda at
      /Users/avioli/repos/RigelEngine/src/game_logic/ai/bomber_plane.cpp:129:5), (lambda at
      /Users/avioli/repos/RigelEngine/src/game_logic/ai/bomber_plane.cpp:144:5)>' requested here
  base::match(mState,
        ^
/usr/local/Cellar/llvm/8.0.0_1/bin/../include/c++/v1/variant:1535:26: note: candidate function [with _Visitor =
      rigel::base::detail::overloaded<(lambda at /Users/avioli/repos/RigelEngine/src/game_logic/ai/bomber_plane.cpp:109:5),
      (lambda at /Users/avioli/repos/RigelEngine/src/game_logic/ai/bomber_plane.cpp:129:5), (lambda at
      /Users/avioli/repos/RigelEngine/src/game_logic/ai/bomber_plane.cpp:144:5)>, _Vs =
      <std::__1::variant<rigel::game_logic::behaviors::BomberPlane::FlyingIn,
      rigel::game_logic::behaviors::BomberPlane::DroppingBomb, rigel::game_logic::behaviors::BomberPlane::FlyingOut> &>] has been
      explicitly made unavailable
constexpr decltype(auto) visit(_Visitor&& __visitor, _Vs&&... __vs) {
                         ^
1 error generated.
make[2]: *** [src/CMakeFiles/rigel_core.dir/game_logic/ai/bomber_plane.cpp.o] Error 1
make[1]: *** [src/CMakeFiles/rigel_core.dir/all] Error 2
make: *** [all] Error 2
```

Re unzipping - you can open a terminal in macOS and do `unzip 4duke.zip` and it will unzip it just fine.

--- a/src/base/warnings.hpp
+++ b/src/base/warnings.hpp
@@ -57,6 +57,7 @@
     _Pragma("clang diagnostic ignored \"-Wfloat-equal\"") \
     _Pragma("clang diagnostic ignored \"-Wgnu-anonymous-struct\"") \
     _Pragma("clang diagnostic ignored \"-Wnested-anon-types\"") \
+    _Pragma("clang diagnostic ignored \"-Wextra-semi-stmt\"") \
     /**/
 
   #define RIGEL_RESTORE_WARNINGS _Pragma("clang diagnostic pop")

--- a/src/data/movie.hpp
+++ b/src/data/movie.hpp
@@ -24,7 +24,6 @@
 namespace rigel { namespace data {
 
 struct MovieFrame {
-  MovieFrame() = default;
   MovieFrame(Image&& replacementImage, const int startRow)
     : mReplacementImage(std::move(replacementImage))
     , mStartRow(startRow)

--- a/src/game_logic/player.cpp
+++ b/src/game_logic/player.cpp
@@ -188,7 +188,7 @@ data::SoundId soundIdForWeapon(const data::WeaponType weaponType) {
 
     default:
       return SoundId::DukeNormalShot;
-  };
+  }
 }
 
 

